packages/state-transition/src/block/processDeposit.ts

```diff
@@ -61,38 +55,43 @@ export function applyDeposit(
   state: CachedBeaconStateAllForks,
   deposit: DepositData | DepositRequest
 ): void {
-  const {config, validators, epochCtx} = state;
-  const {pubkey, withdrawalCredentials, amount} = deposit;
+  const {config, epochCtx, validators} = state;
+  const {pubkey, withdrawalCredentials, amount, signature} = deposit;
 
   const cachedIndex = epochCtx.getValidatorIndex(pubkey);
-  if (cachedIndex === null || !Number.isSafeInteger(cachedIndex) || cachedIndex >= validators.length) {
-    if (isValidDepositSignature(config, pubkey, withdrawalCredentials, amount, deposit.signature)) {
-      addValidatorToRegistry(fork, state, pubkey, withdrawalCredentials, amount);
-    }
-  } else {
-    if (fork < ForkSeq.electra) {
+  const isNewValidator = cachedIndex === null || !Number.isSafeInteger(cachedIndex) || cachedIndex >= validators.length;
```

Comment:
We need to add back condition `cachedIndex >= validators.length` to pass the benchmark check. @twoeths

Reply:
Benchmark seems to pass, is this still an issue?

Reply:
> Benchmark seems to pass, is this still an issue?

It is passing because I added back this condition